pypdf/_crypt_providers/_cryptography.py

```diff
@@ -26,16 +26,22 @@
 # POSSIBILITY OF SUCH DAMAGE.
 
 import secrets
+from ._utils import deprecate_with_replacement
 
 from cryptography import __version__
 from cryptography.hazmat.primitives import padding
 from cryptography.hazmat.primitives.ciphers.algorithms import AES
 
 try:
-    # 43.0.0 - https://cryptography.io/en/latest/changelog/#v43-0-0
     from cryptography.hazmat.decrepit.ciphers.algorithms import ARC4
 except ImportError:
     from cryptography.hazmat.primitives.ciphers.algorithms import ARC4
+    # https://cryptography.io/en/latest/changelog/#v43-0-0
+    deprecate_with_replacement(
+            "cryptography.hazmat.primitives.ciphers.algorithms",
```

Comment:
The user has no control over these directly, thus this does not make much sense here.

Reply:
Remove deprecate_with_replacement and replace with the following message?

`print("The Cryptography library moved the location of ARC4 in v43.0.0. pypdf 6.0.0 will require Cryptography v43.0.0.")`

Reply:
`print` is not intended for user-facing messages. Apart from this, I still do not really see the benefit of dropping the old versions if its only an import and no functionality ...

Reply:
This removal seems to only affect the old variant - the new primary import will not be removed and thus *pypdf* should not break here.

Reply:
Reading their documentation, I think you are right. I would prefer we `deprecate_with_replacement`, in time the number of users who update pypdf and not cryptography will become small. However, since it currently works and they did the namespace move fairly recently, there is reason to keep the same, so feel free to close this.